### PR TITLE
Change default value of the `preferLocal` option to `false`

### PR DESCRIPTION
**Pull request by ehmicky**
This changes the default value of the `preferLocal` option to `false`.

This option causes several issues for some users: #196, #153.

### CODE REVIEW

**Comment by ehmicky** [comment]
CI tests are working now.

**Comment by sindresorhus** [comment]
For posterity, here's my reasoning for doing this change:
- Most cases don't actually spawn local binaries from `node_modules`.
    I realized that most of the cases are global binaries and local ones with `{cwd: __dirname}`.
- The few times you actually need it, it might be more readable to make it explicit.
- It causes surprising behavior then because:
    1. Yarn mocks with the PATH too.
    2. Some Node.js installation has a `node_modules` directory or `npm` directory that will be used instead of what the user actually wanted. This can happen silently and cause confusion.

So basically, I think we should default to the least surprising behavior.
